vertical-pod-autoscaler/enhancements/4831-control-actuation-behavior/README.md

```diff
@@ -0,0 +1,106 @@
+[changing container resources in place](https://github.com/kubernetes/enhancements/issues/1287) as an alpha feature.
+VPA will use that as
+[an alternative way to actuate its recommendations](https://github.com/kubernetes/autoscaler/issues/4016).
+
+We propose adding a new functionality which allows to control the conditions in which recommendations can be applied to
+containers based on the changes (are we scaling up or scaling down?) for each resource individually.
+
+## Motivation
+For some workloads, each eviction introduces disruptions for users. Examples include anything running as a singleton 
+(e.g. etcd) or workload that needs a lot of time to process data on startup (e.g. prometheus). Therefore, operators may
+want to limit the amount of evictions. We can do this by introducing a new tradeoff: Overspend on resources to increase 
+uptime. It is possible to turn off eviction for downscaling entirely or have it managed by an external component to 
+allow downscaling only during certain time windows, days, or based on other criteria.
+
+Scaling containers in place might cause immediate disruption (when containers specify `RestartPolicy: RestartContainer`
+for one of the changed resources) which operators might want to avoid. Even if in place scale down doesn't cause
+immediate disruption there is no guarantee that container will be able to scale up later without eviction. 
+
+### Goals
+* For a specific VPA object, allow turning off actuation for downscaling entirely while still allow for upscaling
+* Allow for resource-specific decisions: The desired policy may be different for CPU and Memory
+
+## Proposal
+Add a new field `ActuationRequirements` to
+[`PodUpdatePolicy`](https://github.com/kubernetes/autoscaler/blob/2f4385b72e304216cf745893747da45ef314898f/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L109)
+of type `[]*ActuationRequirement`. A single `ActuationRequirement` defines a condition which must be `true` to allow
+actuation for the corresponding `Container`. When multiple `ActuationRequirements` are specified for a `Pod`, all of
+them must evaluate to `true` to allow eviction. VPA may partially actuate its recommendation in place, obeying all the
+`ActuationRequirements` while not applying parts of the recommendation that would violate any `ActuationRequirement`.
+
+A single `ActuationRequirement` specifies `Resources` and a `ChangeRequirement` comparing the new recommendation
+(`Target`) with the existing requests on a Pod (`Requests`). Possible values for `Resources` are `[CPU]` and `[Memory]`
+or both `[CPU,Memory]`. If `Resources: [CPU, Memory]`, the condition must be true for either of the two resources to
+allow for actuation. Possible values for `ChangeRequirement` are `TargetHigherThanRequests`,
+`TargetHigherThanOrEqualToRequests`, `TargetLowerThanRequests` and `TargetLowerThanOrEqualToRequests`.
```

Comment:
Why do we need the variations with `OrEqual` ?

Reply:
Given that VPA has other mechanisms in place to ensure there is a large enough diff between the new recommendation and the current requests, I guess it should be fine to remove the `orEqual` variations entirely. WDYT, @jbartosik?

Reply:
Yes, let's do that